### ISSUE TITLE
Does anyone care about plugins removal?

### ISSUE THREAD

**Issue by aobolensk** [Intel-Media-SDK/MediaSDK#2473]
We have decided to remove the following plugins:
```
libmfx_hevcd_hw64.so
libmfx_hevce_hw64.so
libmfx_vp8d_hw64.so
libmfx_vp9d_hw64.so
libmfx_vp9e_hw64.so
```
The reason of removal: implementation of these plugins is already moved to runtime library.
To preserve backward compatibility we are going to fake plugin loading by returning `MFX_ERR_NONE` on load for plugins that are listed above.
Pull request that performs disabling and plugins implementation removal: #2468 

Plugins that are **not** going to be removed:
```
libmfx_h264la_hw64.so
libmfx_hevc_fei_hw64.so
```
Does anyone have any concerns about it?

**Comment by dvrogozh** [comment]
One thing to note: #2468 implements the subj. in 2 steps:
1. Modification to msdk dispatcher to stop loading named plugins, plugin .so are not deleted
2. Dropping plugins .so

So, 1) should not have any negative impact even on apps which has their own dispatcher. But 2) would impact apps which has their own dispatcher. As of now apps with own dispatcher seems to be the only possible issue with this step. Is it? Or I miss something?

**Comment by xhaihao** [comment]
Will you still define MFX_PLUGINID_HEVCD_SW /  MFX_PLUGINID_HEVCD_HW / MFX_PLUGINID_HEVCE_SW / MFX_PLUGINID_HEVCE_HW / MFX_PLUGINID_VP8D_HW / MFX_PLUGINID_VP9E_HW / MFX_PLUGINID_VP9D_HW in the new version of MSDK ? The above variables are used in gst-plugins-bad.

**Comment by dvrogozh** [comment]
> Will you still define MFX_PLUGINID_HEVCD_SW / MFX_PLUGINID_HEVCD_HW

Yes. Nothing changes in header files. If application uses legacy msdk dispatcher (libmfx.so.1) it will still think that it loads a plugin, but dispatcher actually won't load anything and just return OK as if something was loaded. All codec features implemented by plugin still should work since they are implemented in msdk runtime library.

@xhaihao : does msdk gstreamer plugins (from gst-plugins-bad) link against legacy msdk dispatcher or do they have own dispatcher?

**Comment by xhaihao** [comment]
> @xhaihao : does msdk gstreamer plugins (from gst-plugins-bad) link against legacy msdk dispatcher or do they have own dispatcher?`

gst-msdk links against the legacy msdk dispatcher.

**Comment by dmitryermilov** [comment]
> > @xhaihao : does msdk gstreamer plugins (from gst-plugins-bad) link against legacy msdk dispatcher or do they have own dispatcher?`
> 
> gst-msdk links against the legacy msdk dispatcher.

Hi @xhaihao ,

Okay. Is it a gst internal copy of current MediaSDK dispatcher? Or you rely on mfx pkg-config? I'm asking because AFAIR, gst-msdk had an internal copy of MSDK headers. So we want to check what about dispatcher.

**Comment by xhaihao** [comment]
> Okay. Is it a gst internal copy of current MediaSDK dispatcher? Or you rely on mfx pkg-config? I'm asking because AFAIR, gst-msdk had an internal copy of MSDK headers. So we want to check what about dispatcher.

We rely on mfx pkg-config.  gst-msdk in gst-plugins-bad didn't have an internal copy of msdk headers.

**Comment by dvrogozh** [comment]
We just merged #2468 which just disables some plugins loading in msdk dispatcher w/o removing .so themselves.

Next step would be to delete some plugin .so libraries. This is #2490, not yet merged. We install plugins into `/usr/lib/mfx` folder (assuming /usr/lib as a libdir) which is not a folder visible by default by ldconfig. And we name plugin as bare .so libraries w/o specific versioning. All that means that we probably can consider these plugins to be internal msdk stuff, hence can delete them. We will keep this issue hanging for some more time to collect feedback. If nothing will be received, #2490 will be merged.

**Comment by daleksan** [comment]
Hi @dvrogozh do you have any updates for this?
